src/backends/onnx/onnx_backend.cc

```diff
@@ -854,13 +853,6 @@ OnnxBackend::Context::SetInputTensors(
         buffer_offset += expected_byte_sizes[ridx];
       }
 
-#ifdef TRITON_ENABLE_GPU
-      // Synchronize to ensure the buffer is ready to be modified
-      if (string_cuda_copy) {
-        cudaStreamSynchronize(stream_);
-      }
-#endif  // TRITON_ENABLE_GPU
```

Comment:
Why dropping this?

Reply:
If you notice the string_cuda_copy is never used after initializing. It is always false. Thus this is never run

Reply:
Keep `string_cuda_copy` and change line 840 to `string_cuda_copy |= cuda_used;`. FYI, @tanmayv25 this should fix the L0_infer-cudashm

Reply:
Sure. Made this fix.

Reply:
@GuanLuo , I did try that yesterday evening but the backend  still read weird values from the memory. Did you check this change fixed L0_infer_cudashm?

Reply:
I didn't check, just noticed that in this PR. It's weird that the issue is not due to synchronization as it only fails on cudashm variation. I will go over this part more closely once I get to it.